go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -186,17 +190,23 @@ type networkPolicy struct {
 	cancelableContext *util.CancelableContext
 }
 
+type reconcilePeerNamespaces struct {
+	retryNamespaces   *retry.RetryFramework
+	namespaceSelector *metav1.LabelSelector
+}
```

Comment:
The `factory.Handler` already has a `filterFunc` that you can use. I think it might be more beneficial to correlate the `RetryFramework` and `Handler` together, export a method on the `Handler` to get that `filterFunc` so that you can use it.